apiserver/facades/client/resources/repository.go

```diff
@@ -0,0 +1,279 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package resources
+
+import (
+	"context"
+
+	"github.com/juju/charm/v8"
+	charmresource "github.com/juju/charm/v8/resource"
+	csparams "github.com/juju/charmrepo/v6/csclient/params"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/charmhub"
+	"github.com/juju/juju/charmhub/transport"
+	"github.com/juju/juju/charmstore"
+	corecharm "github.com/juju/juju/core/charm"
+)
+
+type NewCharmRepository interface {
+	ResolveResources(resources []charmresource.Resource) ([]charmresource.Resource, error)
+}
+
+// NOTE: There maybe a better way to do this.  Juju's charmhub package is equivalent
+// to charmstore.client.  Juju's charmstore package is what the charmHubClient is doing
+// here, making calls to the charmhub and returning data in a format that the facade
+// would like to see.
+// The question remains, where to put the charmhub piece?
```

Comment:
As we don't have controller locations, I believe here is ok for now.

Reply:
removed